src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/Initializers/WebJobsTelemetryInitializer.cs

```diff
@@ -83,21 +90,24 @@ public void Initialize(ITelemetry telemetry)
             // we may track traces/dependencies after function scope ends - we don't want to update those
             else if (request != null)
             {
+                // set SDK version on all requests to mark that they are collected by webjobs/functions
+                request.Context.GetInternalContext().SdkVersion = sdkVersion;
```

Comment:
> and to expose it via a setting that we really don't want anyone but us to touch
The problem here is that we already want functions host to set it. I understand that you somehow override it on the functions side, but I'm not sure how it works and whether it would work for auto-collected requests.

> Is this not set automatically since it's set on the TelemetryClient?
It's set on the TelemetryClient, and used when webjobs explicitly track telemetry through this client. Auto-collected telemetry is tracked using different instances of the client.
To track usage properly we need to stamp azurefunctions/webjobs on the requests. We probably need this configuration in the AI, so that custom internal version could be propagated to internal TelemetryClients. But for now we can only override whatever AI SDK sets automatically.

Reply:
great idea, one caveat here is that TelemetryClient still has to have sdk version set up so that if customer reports any telemetry through it, it will get the version stamped.

Otherwise, for auto-collected dependencies/metrics/etc (but not requests) we should keep the generated SDK version for debugging.

I also want to make initializer easily replaceable, so it does not have any logic.

Reply:
So one other question here... what TelemetryInitializers do these internal TelemetryClients use? It sounds like they use the same ones as the ones we've registered?